packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -445,4 +458,42 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
             }
         );
     }
+
+    protected async shouldCreateBeforeOpen(uri: TheiaURI): Promise<boolean> {
+        // Do not try to create a document which scheme is not a file
+        if (uri.scheme !== 'file') {
+            return false;
+        }
+        const stat = await this.fileSystem.getFileStat(uri.toString());
+        if (stat && !stat.isDirectory) {
+            return false;
+        }
+        return true;
+    }
+
+    protected async openEditor(uri: URI, columnOrOptions?: ViewColumn | TextDocumentShowOptions): Promise<object | undefined> {
+        let options: TextDocumentShowOptions | undefined;
+        if (typeof columnOrOptions === 'number') {
+            options = {
+                viewColumn: columnOrOptions
+            };
+        } else if (columnOrOptions) {
+            options = {
+                ...columnOrOptions
+            };
+        }
+        const editorOptions = DocumentsMainImpl.toEditorOpenerOptions(this.shell, options);
+        return open(this.openerService, new TheiaURI(uri), editorOptions);
+    }
+
+    protected async getOrCreateDirectory(uri: TheiaURI): Promise<FileStat | undefined> {
+        const stat = await this.fileSystem.getFileStat(uri.toString());
+        if (stat) {
+            if (stat.isDirectory) {
+                return stat;
+            }
+            return this.getOrCreateDirectory(uri.parent);
```

Comment:
Not sure what you're trying to achieve here.

Reply:
In case the URI contains a path to a not existing file that should be placed in some not existing folder tree, then we have to create the required folders before creating the file itself. Don't we?

Reply:
I don't think the logic is right here: if something exists, but it's not a directory we should fail. But anyway I think the node file system will create any required parents of a new directory already, no need to implement this yourself.

Reply:
I was wrong. No need to create the sub-directories in recursion.